autoload/ale/lsp_linter.vim

```diff
@@ -180,6 +180,14 @@ function! ale#lsp_linter#StartLSP(buffer, linter) abort
 
     let l:language_id = ale#util#GetFunction(a:linter.language_callback)(a:buffer)
 
+    if has_key(a:linter, 'lsp_config')
```

Comment:
You can use `if !empty(get(a:linter, 'lsp_config'))` here.

You'll have to update `ale#linter#PreProcess` to accept `lsp_config` as an option, and check the type, and update the tests to check that it's allowed and that it appears in the output okay.

Reply:
Look at what the function does. The `ale#Define` options are validated and new Dictionary values are created from the options. That function hasn't been modified, so this option doesn't work yet.